identify-layer-features/src/main/java/com/esri/arcgismaps/sample/identifylayerfeatures/components/MapViewModel.kt

```diff
@@ -0,0 +1,145 @@
+    val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
+
+    // string text to display the identify layer results
+    val bottomTextBanner = mutableStateOf(String())
+
+    init {
+        // create a feature layer of damaged property data
+        val featureTable = ServiceFeatureTable(application.getString(R.string.damage_assessment))
+        val featureLayer = createWithFeatureTable(featureTable)
+
+        // create a layer with world cities data
+        val mapImageLayer = ArcGISMapImageLayer(application.getString(R.string.world_cities))
+        sampleCoroutineScope.launch {
+            mapImageLayer.load().onSuccess {
+                mapImageLayer.apply {
+                    subLayerContents.value[1].isVisible = false
+                    subLayerContents.value[2].isVisible = false
+                }
+            }.onFailure { error ->
+                // show the message dialog and pass the error message to be displayed in the dialog
+                messageDialogVM.showMessageDialog(error.message.toString(), error.cause.toString())
+            }
+        }
+
+        // create a topographic map
+        val map = ArcGISMap(BasemapStyle.ArcGISTopographic).apply {
+            // add world cities layer
+            operationalLayers.add(mapImageLayer)
+            // add damaged property data
+            operationalLayers.add(featureLayer)
+        }
+        // assign the map to the map view
+        mapViewState.value.arcGISMap = map
+    }
+
+    /**
+     * Identify the feature layer results and display the resulting information
+     */
+    fun handleIdentifyResult(result: Result<List<IdentifyLayerResult>>) {
+        sampleCoroutineScope.launch {
+            result.onSuccess { identifyResultList ->
+                bottomTextBanner.value = ""
+                val message = StringBuilder()
+                var totalCount = 0
+                identifyResultList.forEach { identifyLayerResult ->
+                    val geoElementsCount = geoElementsCountFromResult(identifyLayerResult)
+                    val layerName = identifyLayerResult.layerContent.name
+                    message.append(layerName).append(": ").append(geoElementsCount)
+
+                    // add new line character if not the final element in array
+                    if (identifyLayerResult != identifyResultList[identifyResultList.size - 1]) {
+                        message.append("\n")
+                    }
+                    totalCount += geoElementsCount
+                }
+                // if any elements were found show the results, else notify user that no elements were found
+                if (totalCount > 0) {
+                    bottomTextBanner.value = "Number of elements found:\n${message}"
+                } else {
+                    messageDialogVM.showMessageDialog(
+                        title = "No element found",
+                        description = message.toString()
+                    )
```

Comment:
The `message` would be empty if no elements are found. Maybe use a custom description for the dialog? 

```suggestion
                    messageDialogVM.showMessageDialog(
                        title = "No element found",
                        description = "Tap an area on the map with visible features"
                    )
```